### PR TITLE
Simplify the doctest invocation

### LINKED ISSUE #1431: dhall:doctest doesn't seem to cover all defined doctests
https://github.com/dhall-lang/dhall-haskell/blob/48ca2d242496bc2f1825b198a59c468caa829463/dhall/doctest/Main.hs#L27-L36

This looks like only `Dhall` and `Dhall.{Import,Tutorial,ETags}` are covered. But we also have doctests in `Dhall.Map` and `Dhall.Pretty.Internal`.

### PR DESCRIPTION
**Pull request by sjakobi**
This doesn't change the number of tests covered.

Fixes #1431.

### CODE REVIEW

**Comment by sjakobi** [comment]
Hmm, how did this change cause this failure on Hydra?

```
Test suite doctest: RUNNING...

/tmp/nix-build-dhall-1.26.1.drv-2/dhall-sdist/src/Dhall/Version.hs:9:1: error:
    Could not find module ‘Paths_dhall’
    Use -v to see a list of the files searched for.
  |
9 | import qualified Paths_dhall  as P
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Test suite doctest: FAIL
```

**Comment by sjakobi** [comment]
@quasicomputational Could you review this?

File: dhall/doctest/Main.hs [hunk 1]
```diff
@@ -27,10 +27,17 @@ main = do
...
+            -- The reason is that src/Dhall/Version.hs depends on
+            -- the generated Paths_dhall module which is "out-of-scope"
+            -- when running the testsuite with cabal v1-test.
+            -- Instead, we target a selection of modules whose combined module
+            -- dependency tree covers all modules that contain doctests.
+
+            -- , prefix </> "src"
             , "-i" <> (prefix </> "src")
-            , "-i" <> (prefix </> "ghc-src")
-            , prefix </> "src/Dhall.hs"
```
Comment (Gabriel439) [inline dhall/doctest/Main.hs]: Is the new version still testing the `Dhall` and `Dhall.Import` modules?

Comment (sjakobi) [reply dhall/doctest/Main.hs]: Yes. `Dhall.Tutorial` depends on `Dhall`, which depends on `Dhall.Import`.

Comment (Gabriel439) [reply dhall/doctest/Main.hs]: But does testing `Dhall.Tutorial` guarantee that `Dhall` is also tested?

Comment (sjakobi) [reply dhall/doctest/Main.hs]: That is my current understanding. At least this logic explains why `Dhall.Map` was being tested. @quasicomputational can probably clarify.

And it seems that I should then clarify the comment above.

**Comment by quasicomputational** [comment]
I'm still pretty puzzled by all this; `v1-test` not paying well with `Paths_*` is a new one on me. Does #1125 obsolete the comment added?

**Comment by sjakobi** [comment]
> `v1-test` not paying well with `Paths_*` is a new one on me.

It might be a Nix-specific problem? I inferred it from the first Hydra results.

> Does #1125 obsolete the comment added?

If it works out, sure!

**Pull request merged**